src/content-helper/excerpt-generator/components/excerpt-panel.tsx

```diff
@@ -144,10 +189,24 @@ const PostExcerptGenerator = () => {
 					__nextHasNoMarginBottom
 					label={ __( 'Write an excerpt (optional)', 'wp-parsely' ) }
 					className="editor-post-excerpt__textarea"
-					onChange={ ( value ) => editPost( { excerpt: value } ) }
-					readOnly={ isLoading || hasGeneratedExcerpt }
+					onChange={ ( value ) => {
+						setOnChangeFired( true );
+						editPost( { excerpt: value } );
+						setExcerptData( { ...excerptData, currentExcerpt: value } );
```

Comment:
If we change the generated excerpt and not approve it, nor discard it, will it not still be saved to the post? I think the `editPost` function should only be called when `excerptData.newExcerptGenerated` is false/empty.

Reply:
Do you think that this saving would take place due to `editPost()`? In my testing, the only way to persist the excerpt is to press the post's `Update` button. `editPost()` doesn't seem to have a permanent effect.

Reply:
Ah yeah, that sounds right! This would only happen if the user wouldn't reject nor accept the excerpt, but would still save the post. So maybe not really an issue, I will leave that to your discretion 🙁

Reply:
There's an exception. If the user has typed something manually in that state, changes persist.

Reply:
So there's a potential for unexpected behavior. I'll leave this open for the time being.

Reply:
This should be fixed now.